TEST_SCENARIOS.md

```diff
@@ -0,0 +1,73 @@
+# End-to-end test scenarios
+
+These can be performed manually and/or automated. More scenarios will be added as they are implemented (XMTP v3 is a work in progress).
+
+## How to read the tests
+
+### Identifiers
+
+- Accounts (in this case wallets) are identified by letters (A, B, C)
+- Installations are identified by numbers (1, 2, 3)
+
+For example, Client A1 is the first installation of account A, A2 is the second installation of account A, and B1 is the first installation of account B.
+
+### Registering
+
+Each 'register' step below creates an installation and associates it with the relevant account.
+
+If performed against the CLI, multiple installations can be associated with the same wallet by generating a random number for the wallet and specifying it as a `--seed` param to the `register` command. Care must be taken to ensure a brand new seed is used for each invocation of the test so that previous test runs do not interfere.
+
+### Verifying
+
+Each 'verify' step below involves listing the conversations and messages for a given installation.
+
+Each installation is expected to have the history of all messages sent or received from the moment it was registered onwards. Currently, messages sent before a specific installation was registered are not expected to be viewable on that installation.
+
+Message history is expected to persist between cold starts and does not require a network connection to access.
+
+## Test cases
+
+### 1. Sending and receiving as installations are added
+
+Clients are expected to detect and establish sessions with new installations as they register.
+
+```
+Register client A1 and B1
+    Send a message from A1 to B
+    Verify message was received in all clients
+    Respond from B1 to A
+    Verify message was received in all clients
+Register client A2
+    Send a message from A1 to B
+    Verify message was received in all clients
+Register client B2
+    Send a message from A1 to B
+    Verify message was received in all clients
+Register client A3
+    Send a message from A3 to B
+    Verify message was received in all clients
+```
```

Comment:
I like this formulation

Test:
- the nominal case
- That clients detect new self-installations
- That clients fanout to new remote-installations
- New installations can be heard. 


Think this is fantastic scope